examples/ct/structures.go

```diff
@@ -99,15 +112,208 @@ func SignV1SCTForPrecertificate(km crypto.KeyManager, cert *x509.Certificate, t
 	keyHash := sha256.Sum256(cert.RawSubjectPublicKeyInfo)
 	precert := ct.PreCert{IssuerKeyHash: keyHash, TBSCertificate: cert.RawTBSCertificate}
 
-	timestampedEntry := ct.TimestampedEntry{Timestamp:sctInput.Timestamp, EntryType:ct.PrecertLogEntryType, PrecertEntry:precert}
-	leaf := ct.MerkleTreeLeaf{Version: ct.V1, LeafType:ct.TimestampedEntryLeafType, TimestampedEntry:timestampedEntry}
+	timestampedEntry := ct.TimestampedEntry{Timestamp: sctInput.Timestamp, EntryType: ct.PrecertLogEntryType, PrecertEntry: precert}
+	leaf := ct.MerkleTreeLeaf{Version: ct.V1, LeafType: ct.TimestampedEntryLeafType, TimestampedEntry: timestampedEntry}
 
 	return serializeAndSignSCT(km, leaf, sctInput, t)
 }
 
-func getSCTForSignatureInput(t time.Time) (ct.SignedCertificateTimestamp) {
+func getSCTForSignatureInput(t time.Time) ct.SignedCertificateTimestamp {
 	return ct.SignedCertificateTimestamp{
-		SCTVersion:ct.V1,
+		SCTVersion: ct.V1,
 		Timestamp:  uint64(t.UnixNano() / millisPerNano), // spec uses millisecond timestamps
 		Extensions: ct.CTExtensions{}}
-}
\ No newline at end of file
+}
+
+func NewLogEntry(leaf ct.MerkleTreeLeaf, certChain []*x509.Certificate) *CTLogEntry {
```

Comment:
Validation is done by the request handler before this gets created. The rest of the code works with x509.Certificate objects. This is only so we can serialize the extra data that the backend needs to store so we can implement CT on the generic storage.